### ISSUE TITLE
Number of workers set to 0 when instantiating MainCircuitBreaker

### ISSUE THREAD

**Issue by tiggris** [rebus-org/Rebus.CircuitBreaker#6]
### Issue description

It seems that when registration of MainCircuitBreaker is called the number of workers in resolved Options instance is set to 0:

```
 configurer.Register(context =>
{
    var loggerFactory = context.Get<IRebusLoggerFactory>();
    var asyncTaskFactory = context.Get<IAsyncTaskFactory>();
    var circuitBreakerEvents = context.Get<CircuitBreakerEvents>();
    var options = context.Get<Options>();
    var circuitBreakers = builder.Build(context);

    return new MainCircuitBreaker(circuitBreakers, loggerFactory, asyncTaskFactory, new Lazy<IBus>(context.Get<IBus>), circuitBreakerEvents, options);
});
```

Sample Rebus configuration (used for testing purposes):

```
services.AddRebus(configure =>
{
    return configure
        .Transport(t =>
        {
                t.UseAzureServiceBus(...)
        })
        .Options(options =>
        {
            options.SetNumberOfWorkers(1);
            options.SetMaxParallelism(1);
            options.EnableCircuitBreaker(config => config.OpenOn<WebRequestException>(1, 5, 10, 20));                            
        });
});
```

I suppose this behavior is caused by IBusStarter.Create() method being called before the above code is executed.

### Provider and version information

Rebus.CircuitBreaker version: 0.0.2
Target framework: .Net Core 5.0
Queue provider: Azure Service Bus
Operating system: Windows 10
IDE: Visual Studio 2022 Preview

**Comment by tiggris** [comment]
Looking deeper into the issue it looks like options.NumberOfWorkers is never set back to the original value so postponing of reading it won't change anything. With the current implementation of Rebus it seems that the only option is to read and initialize number of workers in MainCircuitBreaker.Trip() method:

```
public async Task Trip(Exception exception)
{
    if (_configuredNumberOfWorkers == 0)
    {
        _configuredNumberOfWorkers = _bus.Value.Advanced.Workers.Count;
    }

    await InvokeCircuitBreakerAction(circuitBreaker => circuitBreaker.Trip(exception));
}
```

**Comment by NickyNest** [comment]
I've faced with the same issue.
```
                    .Options(options =>
                    {
                       ...
                        options.SetNumberOfWorkers(2);
                        options.EnableCircuitBreaker(c =>
                        {
                            c.OpenOn<InvalidOperationException>(attempts: 3, trackingPeriodInSeconds: 20,
                                halfOpenPeriodInSeconds: 30, resetIntervalInSeconds: 60);
                        });
                    })
```
![image](https://user-images.githubusercontent.com/25012435/153040129-c961d3ff-6983-4e2b-9e1c-51e64e4b4c1c.png)
![image](https://user-images.githubusercontent.com/25012435/153040278-8591a600-23a9-4c91-adce-6e2fd7cc1060.png)
Looks like CircuitBreaker is started before options are initialized.

**Comment by jwoots** [comment]
A pull request for this issue here : https://github.com/rebus-org/Rebus.CircuitBreaker/pull/9

**Comment by mookid8000** [comment]
Fixed by #9 - it's out as Rebus.CircuitBreaker 0.0.4 on NuGet.org now 🙂
